### ISSUE TITLE
Steamwrap?

### ISSUE THREAD

**Issue by larsiusprime** [HaxeFoundation/Project-Management#50]
I recently took it upon myself to upgrade the [steamwrap](http://github.com/larsiusprime/steamwrap) library, adding steam controller support and adding CFFI prime bindings where possible.

I wanted to see what interest the Haxe Foundation has in this library, if any.

Possible reasons for involving HF:
- Currently Steamwrap supports Lime/OpenFL (and theoretically NME) only, but with some guidance I would like to see if I can make it framework agnostic.
- Properly testing this requires using an actual steam app with a legitimate steam app ID. Right now all I can test it with is Defender's Quest, which does not represent a comprehensive coverage test of the Steam API. If the Haxe Foundation had a relationship with steam they could perhaps secure an official Haxe test application app id and steam account that we could rig up to a continuous integration server (perhaps running steamOs or ubuntu or something). Needless to say, this is beyond my current abilities.
- Also, having an official relationship between Valve and HF could be good for developer outreach and awareness of Haxe.
- I'm currently maintaining this at my own pace and happy to keep it alive, but without external feedback it will basically just be a list of features I need for my own game, perhaps not appropriate for general community use.
- If we want this to support targets other than C++ (say C# target for use with Unity+Haxe games), then I will need help with that.

If HF is not interested in this, that's fine, I'll just keep maintaining it on my own :)

**Comment by ncannasse** [comment]
While as you must know I'm personally interested in it, I don't think HF currently has any action to make in that direction. We're trying to keep the number to things we maintain at a sustainable level and that can't include things as much specific as steam support until we have a larger team and more financing :)

**Comment by larsiusprime** [comment]
Makes sense! We can table this until a later date; in the meantime maybe some of us Haxe steam developers could get together and talk about whether we want to do some of these things unofficially

**Comment by larsiusprime** [comment]
(Feel free to close unless you're waiting for feedback from others)

**Comment by Simn** [comment]
Let's keep it open for a bit, maybe some interesting feedback comes along.

**Comment by delahee** [comment]
I was thinking that maybe we should at least put the haxe installer in the
steam third party tools like spriter and the likes maybe ?
Le 24 févr. 2016 21:15, "Simon Krajewski" notifications@github.com a
écrit :

**Comment by larsiusprime** [comment]
@delahee Do you mean contacting steam to get an APP ID for "install haxe" itself, or some such thing? And then "Haxe" is a free app on Steam that anyone can just go get?

**Comment by Simn** [comment]
I don't think that's appropriate. For a proper framework maybe, but Haxe itself?

**Comment by larsiusprime** [comment]
It could be a way to get more exposure, but the way steam installs things doesn't necessarily work well with installing a basic dev environment like haxe... what would happen when you push an updated version of the binary through steampipe? What would the user experience be? Would steam "own" the haxe installation folder, or would it just be a delivery vehicle for the installer binary?

**Comment by delahee** [comment]
Well you're part right even if IMHO haxe/flash itself is a fully fledged
framework... Or maybe we can pack a "haxe+" release auto installing
starling, openfl, kha and pixie, whatever..

I think we have strength in game biz, why not play the card full throttle ?
Le 24 févr. 2016 21:58, "Simon Krajewski" notifications@github.com a
écrit :

**Comment by larsiusprime** [comment]
It sounds like an interesting idea but it would need someone prepared to support it and maintain it and keep the versions up to date, and also manage expectations from the sudden influx of users who find it on steam (also consider they might be much more newbie than we're used to).

Interested to hear what other people think of this besides us four.

**Comment by larsiusprime** [comment]
Wait -- here's a thought. We could approach Valve about joining the Haxe Foundation :P, see if they want to actively support us! More cross platform games on mac/win/steamos is good for them right?

Longshot, of course.

**Comment by Simn** [comment]
I like Valve, but that really is a longshot on overwatch while disoriented.
